scripts/dev_fd_quadrature.jl

```diff
@@ -0,0 +1,118 @@
+#with 10 pts using the T/30 30T range get 1e-3 error, will do this for testing
+
+# CC quadrature
+ncc = n+3
+@time xGCC, wGCC = gausslobatto( ncc )
+@time IGCC = quad_sum(xGCC,wGCC,ft)
+println("Relative error on CC: ", abs.(1-IGCC/answer), "  with n= ", ncc," points.")
+#30 pts is pretty good - relative tolerance of 1e-6 for Leg, 1e-4 for cheby zeros
+
+#Use asymptotic expansion instead of just truncating the integral (standard practice)
+
+#FIXME: Need to take into account possible integration over splines instead of f
+#If integrate over spline is faster but probably introduces interp error, especially at ends
+#Also we won't spline the perts - but should check this though for bg
+
+# More realistic test on perts
+#Look at a the result of the un-normalized density integral at z=0 using dlnf0dlnq to test q-dep
+#"truth" as a higher-order low-tolerance quadgk over larger q-range
+dlnf0dlnq0(q) = -q / T /(1 + exp(-q/T))
+
+#high-tolerance QuadGK
+a=1#10^-4
+@time Iqgk_truth = quadgk(q ->  √(q^2 + a^2 * 0.06^2)* q^2 * f0(q)*dlnf0dlnq0(q) ,10.0 .^logqmin, 10.0 .^logqmax,rtol=1e-12)[1]
+
+#fast QuadGK at 1e-3
+@time Iqgk_fast = quadgk(q ->  √(q^2 + a^2 * 0.06^2)* q^2 * f0(q)*dlnf0dlnq0(q) ,10.0 .^logqmin, 10.0 .^logqmax,rtol=1e-3)[1]
+println("Relative error on realistic 1e-3 gk: ", abs.(1-Iqgk_fast/Iqgk_truth))
+
+
+#vs FastGaussQuadrature
+function ftp(x)
+    return (√((10.0 ^ from_ui(x,logqmin,logqmax))^2 + a^2 * 0.06^2)
+            * f0(10.0 ^ from_ui(x,logqmin,logqmax)) * (10.0 ^ from_ui(x,logqmin,logqmax))^2
+            *dlnf0dlnq0(10.0 ^ from_ui(x,logqmin,logqmax)) )/ dxdq(10.0 ^ from_ui(x,logqmin,logqmax))
+end
+
+#Gauss-Legendre
+m = 15
+@time xGLe2, wGLe2 = gausslegendre( m )
+@time IGLe2 = quad_sum(xGLe2,wGLe2,ftp)
+println("Relative error on realistic GLeg: ", abs.(1-IGLe2/Iqgk_truth), "  with m= ", m," points.")
+
+#CC
+mcc = m#+1
+@time xGCC2, wGCC2 = gausslobatto( mcc )
+@time IGCC2 = quad_sum(xGCC2,wGCC2,ftp)
+println("Relative error on CC: ", abs.(1-IGCC2/Iqgk_truth), "  with n= ", mcc," points.")
+
+#comparing to the f0 test, where we got 1e-3 error for 10 and 13 pts for GL and CC respectively, here we have
+#12 and 13 pts, respectively, and similarly, we have that it is faster
+#Need to do a careful comparison at some point where we can average over a number of times,
+#as it is hard to know when the quad points are reusing pre-computed values...
+#Either way so long as the computation of quadrature points (which occurs in the background once)
+#plus the sum operations that happen frequently are faster than quadgk * number of times it is called,
+#then we are fine - and I think this is an easily attainable threshold for a small number of quad pts (<15)
+
+#Would be interesting to see if cosmology affects this accuracy, because could reduce cost even further by
+#just passing the q-pts (computed externally) once instead of computing in bg
+
```

Comment:
Another choice would have been to do the integral in q using Gauss-Laguerre (CAMB claims to be able to do it with 4 pts???) -- seems like exp q (is the reference Bond & Szalay?) is working well here, though.